src/backends/tensorrt/tensorrtlib.cc

```diff
@@ -789,6 +789,14 @@ namespace dd
         if (num_processed == 0)
           break;
 
+        // some models don't support dynamic batch size (ex: onnx)
+        // this can lead to undetected bad predictions
+        if (num_processed > _dims.d[0])
```

Comment:
@Bycob don´t we want to limit this to yolox models ? Usually, if batch size > compiled batch size, tensorrt throws a warning saying that performances aren´t optimal, which is enough.

Reply:
Here we check the batch sent by the input connector, so for caffe model this would never trigger, since the model is compiled with the correct batch size. This check is for when max batch size (specified by API) is bigger than the onnx model batch size. Ideally we should check that at service creation, but we only read the onnx model at the first inference